### PR TITLE
MultiMix Env

### PR DESCRIPTION
**Pull request by Tezirg**
Adds `MultiMixEnv` class as a wrapper of multiple environments (on the same grid).
The goal is to use it in the context of MultiMix training, where the BaseEnvironment attributes and class attributes are accessible through the wrapper in a transparent fashion.

### CODE REVIEW

File: grid2op/Environment/MultiMixEnv.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        return getattr(self.current_env, name)
+
+    def reset(self):
+        rnd_env_idx = np.random.randint(len(self._envs))
+        self.current_env = self._envs[rnd_env_idx]
+        self.current_env.reset()
+        return self.get_obs()
+
+    def seed(self, seed=None):
+        """
```
Comment (BDonnot) [inline grid2op/Environment/MultiMixEnv.py:76]: This should rather define as many seeds as there are element in _envs, and call the seed method on all of them.

File: grid2op/Environment/MultiMixEnv.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        self.current_env = self._envs[0]
+        # Make sure GridObject class attributes are set from first env
+        # Shouldbe fine since the grid is the same for all envs
+        self.__class__ = self.init_grid(self.current_env)
+
+    def __getattr__(self, name):
+        return getattr(self.current_env, name)
+
+    def reset(self):
+        rnd_env_idx = np.random.randint(len(self._envs))
```
Comment (BDonnot) [inline grid2op/Environment/MultiMixEnv.py]: Maybe just loop through the _envs deterministically and add a possibility to loop randomly. More control is always better.

File: grid2op/Environment/MultiMixEnv.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+    def set_thermal_limit(self, thermal_limit):
+        """
+        Set the thermal limit effectively.
+        Will propagate to all underlying environments
+        """
+        for e in self._envs:
+            e.set_thermal_limit(thermal_limit)
+
+    def get_obs(self):
```
Comment (BDonnot) [inline grid2op/Environment/MultiMixEnv.py]: Not necessary with the overload of __getattr___ i believe

File: grid2op/Environment/MultiMixEnv.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+        if len(self._envs) == 0:
+            err_msg = "MultiMix envs_dir did not contain any valid env"
+            raise EnvError(err_msg)
+            
+        self.current_env = self._envs[0]
+        # Make sure GridObject class attributes are set from first env
+        # Shouldbe fine since the grid is the same for all envs
+        self.__class__ = self.init_grid(self.current_env)
+
```
Comment (BDonnot) [inline grid2op/Environment/MultiMixEnv.py:57]: Add a method to get the current env ID maybe?

**Review by BDonnot — CHANGES_REQUESTED** [review]
A few modifications to the base multienv class.
Only the seed is really a requirement, the current implementation does not guarantee reproducibility I think.

File: grid2op/Environment/MultiMixEnv.py (hunk 3 above)
Comment (Tezirg) [reply grid2op/Environment/MultiMixEnv.py]: True, same goes for get_thermals and step. Removed these.

File: grid2op/Environment/MultiMixEnv.py (hunk 4 above)
Comment (Tezirg) [reply grid2op/Environment/MultiMixEnv.py:57]: Aye aye sir, good idea

**Review by BDonnot — APPROVED** [review]

**Pull request merged**